ftl-build/aarch64/Dockerfile

```diff
@@ -67,6 +68,15 @@ RUN curl -sSL https://ftl.pi-hole.net/libraries/nettle-${nettleversion}.tar.gz |
     && cd .. \
     && rm -r nettle-${nettleversion}
 
+# Build static mbedTLS with pthread support
+RUN curl -sSL https://ftl.pi-hole.net/libraries/mbedtls-${mbedtlsversion}.tar.gz | tar -xz \
```

Comment:
Can we get this directly from Mbed-TLS/releases instead? I'm more comfortable with getting it straight from the source.

Also, looks like there's a major version update at https://github.com/Mbed-TLS/mbedtls/releases/tag/v3.4.0

Reply:
> Can we get this directly from Mbed-TLS/releases instead? I'm more comfortable with getting it straight from the source.

We did that in the past but have been affected several times with servers being down / certificates expired / other issues which ultimately prevented us from building any more Pi-hole docker images. Hence, we decided to mirror them on `ftl.pi-hole.net`. I don't think this is an issue as the packages are unmodified and any curious user can download them all, download them from the primary source and checksum them. They will all match. Also, I'd like to avoid making the `Dockerfile` complex by adding something like "`download from `ftp.gnu.org`, if this fails, try again from `ftl.pi-hole.net`".

> Also, looks like there's a major version update at https://github.com/Mbed-TLS/mbedtls/releases/tag/v3.4.0

Yes, but CivetWeb isn't compatible with v3.x so far. I plan to do the migration (and contribute this upstream) but this is something likely for after the v6.0 release. The v2.x branch lives on. We're using the most recent v2.x version here which is also a [long-time support branch](https://github.com/Mbed-TLS/mbedtls/releases/tag/v2.28.3). It will be supported with bug-fixes and security fixes until end of 2024 giving us enough time to do the migration. Maybe someone else will do it meanwhile saving us some work: https://github.com/civetweb/civetweb/issues/1143

Reply:
I ported `civetweb`'s `mbedtls` extension to be compatible to the mbed TLS v3.x release series (https://github.com/civetweb/civetweb/pull/1144)

Reply:
> We did that in the past but have been affected several times with servers being down / certificates expired / other issues which ultimately prevented us from building any more Pi-hole docker images.

I trust Github to be up and working more than I trust `ftl`. These docker images shouldn't be built that often and even if they can't be built then they shouldn't impact building the Pi-hole docker image. No need to use `gnu.org` anyways, the tarballs are in the `https://github.com/Mbed-TLS/mbedtls/releases/`.

And if GitHub is down then none of these images could be built via GitHub Actions anyways. :)

Reply:
I was referring to other packages we already have (`libidn` and friends) which are on less reliable servers. I changed the mbed TLS dependency to download from Github now.